### ISSUE TITLE
Crash on CSS bundling with `@media` rule in canary

### ISSUE THREAD

**Issue by nobkd** [oven-sh/bun#14537]
### How can we reproduce the crash?

`test1.css`:
```css
@media (max-width: 700px) {}
```

run:
```sh
bun build --experimental-css test1.css
```

---

The same code still works on stable v1.1.30 (update: >=v1.1.31 fails):

```css
/* test1.css */
@media (width <= 700px) {
  
}
```

### Relevant log output
```
============================================================
Bun Canary v1.1.31-canary.37 (47ff4748) Linux x64
Linux Kernel v6.10.12 | glibc v2.39
CPU: sse42 popcnt avx avx2 avx512
Args: "bun" "build" "--experimental-css" "test1.css"
Elapsed: 6ms | User: 0ms | Sys: 7ms
RSS: 1.07GB | Peak: 15.97MB | Commit: 1.07GB | Faults: 0

panic: Segmentation fault at address 0x0
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.31/lb247ff474AA+vmQ0pni3Co7+5mD4lor3Cqu/mhDst7lB2g7mCA2AA

[1]    62655 illegal hardware instruction (core dumped)  bun build --experimental-css test1.css
```

### Stack Trace (bun.report)

Bun v1.1.31-canary ([`47ff474`](<https://github.com/oven-sh/bun/tree/47ff4748bd03aef9583b8d7bf9d503a8f3db599d>)) on linux x86_64 [BuildCommand]

Segmentation fault at address 0x00000000

- `??`
- [`css_parser.zig:2646`](<https://github.com/oven-sh/bun/blob/47ff4748bd03aef9583b8d7bf9d503a8f3db599d/src/css/css_parser.zig#L2646>): `src.css.css_parser.StyleSheet`
- [`bundle_v2.zig:3238`](<https://github.com/oven-sh/bun/blob/47ff4748bd03aef9583b8d7bf9d503a8f3db599d/src/bundler/bundle_v2.zig#L3238>): `src.bundler.bundle_v2.ParseTask.getAST`
- [`bundle_v2.zig:3443`](<https://github.com/oven-sh/bun/blob/47ff4748bd03aef9583b8d7bf9d503a8f3db599d/src/bundler/bundle_v2.zig#L3443>): `src.bundler.bundle_v2.ParseTask.callback`
- [`thread_pool.zig:680`](<https://github.com/oven-sh/bun/blob/47ff4748bd03aef9583b8d7bf9d503a8f3db599d/src/thread_pool.zig#L680>): `Thread.PosixThreadImpl.spawn__anon_115760.Instance.entryFn`

**Comment by noClaps** [comment]
I'm still seeing this on v1.1.33.

My code:

```css
/* test.css */
@media (min-width: 768px) {
    header {
        background-color: red;
    }
}
```

### Relevant log output

```
============================================================
Bun v1.1.33 (247456b6) macOS Silicon
macOS v15.0.1
Args: "bun" "build" "test.css" "--experimental-css" "--minify"
Features: tsconfig
Elapsed: 2ms | User: 6ms | Sys: 4ms
RSS: 10.88MB | Peak: 10.86MB | Commit: 1.07GB | Faults: 17

panic: Segmentation fault at address 0xAAAAAAAAAAAAAAA2
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.33/Mb1247456bAggggggC_+ttiB+ttiBmm/xJu7+pB+1ywJ__A2t1q1q1C91q1q1C

zsh: trace trap  bun build test.css --experimental-css --minify
```

### Stack trace

Bun v1.1.33 ([`247456b`](<https://github.com/oven-sh/bun/tree/247456b675d4bf4b9cd920e0a080b7002325c0df>)) on macos aarch64 [BuildCommand]

Segmentation fault at address 0xAAAAAAAAAAAAAAA2

- *1 unknown/js code*
- `src.resolver.resolve_path.relativePlatform__anon_96944`
- `src.resolver.resolve_path.relativePlatform__anon_96944`
- `src.css.css_parser.StyleSheet`
- `src.bundler.bundle_v2.ParseTask.getAST`
- `src.bundler.bundle_v2.ParseTask.callback`
- `Thread.PosixThreadImpl.spawn__anon_116617.Instance.entryFn`

Features: tsconfig

**Comment by noClaps** [comment]
Upon further testing, I discovered that

```css
.link {
    &:hover {
        text-decoration: none;
    }

    @media (max-width: 800px) {
        grid-template-columns: none;
        grid-template-rows: auto;
    }
}
```

seems to build fine, but

```css
.link {
    @media (max-width: 800px) {
        grid-template-columns: none;
        grid-template-rows: auto;
    }
}
```

doesn't. It seems that bundling with media queries only works if there is some other nested element or pseudo-class/element  inside the parent selector. Having just regular properties, like:

```css
.link {
    text-decoration: none;

    @media (max-width: 800px) {
        grid-template-columns: none;
        grid-template-rows: auto;
    }
}
```

doesn't seem to work either.

**Comment by zackradisic** [comment]
This has been fixed by #14832 

Please let us know if the issue persists.
